### PR TITLE
detect ipv6 dest addr and pass with [] to rsync

### PR DESCRIPTION
**Pull request by tesshuflower**
Signed-off-by: Tesshu Flower <tflower@redhat.com>

**Describe what this PR does**
Rsync if provided with an IPv6 address needs to use the address in the format [ipv6addr].  This is a URL format.

In the mover script, attempt to detect if the destination address is an ipv6 address and wrap with [] only if it does not already contain [].  This hopefully avoids issues with urls that are already formatted correctly.

**Is there anything that requires special attention?**
- Crude ipv6 check - checking that there are enough stanzas with : (or :: is found)
- Also avoids wrapping with [] if the address already contains [] - potentially could happen with a url address that embeds an ipv6 address?

**Related issues:**
https://github.com/backube/volsync/issues/402

### CODE REVIEW

File: mover-rsync/source.sh [hunk 1]
```diff
@@ -39,19 +39,27 @@ Host *
   TCPKeepAlive no
 SSHCONFIG
 
+URL_DESTINATION_ADDRESS=$DESTINATION_ADDRESS
+if [[ "$DESTINATION_ADDRESS" == *"::"* || "$DESTINATION_ADDRESS" == *":"*":"*":"*":"*":"*":"*":"* ]]; then
+  echo "Destination address $DESTINATION_ADDRESS is ipv6"
+  if [[ "$DESTINATION_ADDRESS" != *"["*"]"* ]]; then
+    URL_DESTINATION_ADDRESS="[$DESTINATION_ADDRESS]"
+  fi
+fi
```
Comment (JohnStrunk) [inline mover-rsync/source.sh:48]: The matching makes me nervous. What about bash regular expressions:
```bash
# Looking for either:
# 1) 8 groups of hex digits separated by ":"
# 2) a "::" in the string
IPV6_REGEX='(([0-9a-fA-F]{0,4}:){7}[0-9a-fA-F]{0,4})|(::)'

if [[ "$DESTINATION_ADDRESS" =~ $IPV6_REGEX ]]; then
    echo "It's IPv6"
fi

if [[ ! "$DESTINATION_ADDRESS" =~ \[.*\] ]]; then
    echo "no brackets"
fi
```

Comment (tesshuflower) [reply mover-rsync/source.sh:48]: That's much nicer than what I had - I'm terrible with regex's so threw my version together fast before I lost that ipv6 env...I've done a bunch of different address checks and I think your suggestion looks good to me - will update...

**Pull request merged**